Extract library functionality from gradle plugin

hemarkus opened the pull request:
Keep only gradle parts in the plugin

Commit:
Extract library functionality from gradle plugin

Keep only gradle parts in the plugin

GiasemiSh commented on build.gradle:
```diff
@@ -25,13 +25,13 @@ sourceCompatibility = 1.8
 
 group = "org.qmstr"
-version = "0.3.0"
+version = "0.3.1"
```
This should be updated once it is merged back to the qmstr repo

GiasemiSh reviewed (changes requested):
Please skip the update of the version

Commit:
reset version

version will be set from the versions.env file when module moves back to main qmstr repo

Commit:
reset version
